data/migrations/20190329111439_projects_table.js

```diff
@@ -0,0 +1,42 @@
+exports.up = function(knex, Promise) {
+  return knex.schema
+    .createTable("projects", tbl => {
+      tbl.increments();
+      tbl
+        .string("name", 128)
+        .notNullable()
+        .unique();
+      tbl
+        .string("description", 128)
+        .notNullable()
+        .unique();
+      tbl
+        .boolean("complete")
+        .notNullable()
+        .defaultTo(false);
+    })
+    .createTable("actions", tbl => {
+      tbl.increments();
+      tbl
+        .integer("project_id")
+        .unsigned()
+        .notNullable()
+        .references("id")
+        .inTable("projects")
+        .onDelete("CASCADE")
+        .onUpdate("CASCADE");
+      tbl
+        .string("description", 128)
+        .notNullable()
+        .unique();
+      tbl.string("notes", 128).unique();
+      tbl
+        .boolean("complete")
+        .notNullable()
+        .defaultTo(false);
+    });
+};
+
+exports.down = function(knex, Promise) {
+  return knex.schema.dropTableIfExists("actions").dropTableIfExists("projects");
+};
```

Comment:
I like how you put two tables in this single migration. I will have to try this out sometime!